java/fury-core/src/main/java/org/apache/fury/serializer/EnumSerializer.java

```diff
@@ -20,40 +20,69 @@
 package org.apache.fury.serializer;
 
 import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
 import org.apache.fury.Fury;
 import org.apache.fury.memory.MemoryBuffer;
+import org.apache.fury.meta.MetaString;
+import org.apache.fury.resolver.MetaStringResolver;
 import org.apache.fury.util.Preconditions;
 
 @SuppressWarnings("rawtypes")
-public final class EnumSerializer extends ImmutableSerializer<Enum> {
+public class EnumSerializer extends ImmutableSerializer<Enum> {
+  private static final MetaStringResolver META_STRING_RESOLVER = new MetaStringResolver();
   private final Enum[] enumConstants;
+  private final Map<String, Enum> enumStringRepresentation;
```

Comment:
If key is MetaStringBytes, could you use IdentityMap instead?

Reply:
Hi @chaokunyang , thank you for your review.
I have tried to use IdentityMap as you mentioned above, but somehow it fail to match the key. do i missed something here?
<img width="825" alt="image" src="https://github.com/user-attachments/assets/cdedcb63-8357-4933-bebe-b0139b410e4b">